File: Frontend/fpams-web/src/types/appraisalForm.ts
export type AppraisalStatus =
    | "Draft"
    | "Submitted"
    | "HodReviewed"
    | "PrincipalReviewed"
    | "IqacReviewed"
    | "Returned";

export interface AppraisalForm {
    id: string;
    facultyProfileId: string;
    facultyName: string;
    employeeCode: string;
    departmentName: string;
    academicYearId: string;
    academicYearName: string;
    teachingScore: number;
    researchScore: number;
    administrationScore: number;
    contributionScore: number;
    totalScore: number;
    evidenceSummary: string;
    facultyRemarks: string;
    reviewerRemarks: string;
    status: AppraisalStatus;
    submittedOn?: string | null;
}

export interface CreateAppraisalFormRequest {
    facultyProfileId: string;
    academicYearId: string;
    teachingScore: number;
    researchScore: number;
    administrationScore: number;
    contributionScore: number;
    evidenceSummary: string;
    facultyRemarks: string;
}

export interface UpdateAppraisalFormRequest extends CreateAppraisalFormRequest {
    id: string;
}

File: Backend/FPAMS/FPAMS.Domain/Entities/AppraisalForm.cs
using FPAMS.Domain.Common;

namespace FPAMS.Domain.Entities;

public class AppraisalForm : BaseEntity
{
    public Guid FacultyProfileId { get; set; }

    public FacultyProfile FacultyProfile { get; set; } = null!;

    public Guid AcademicYearId { get; set; }

    public AcademicYear AcademicYear { get; set; } = null!;

    public int TeachingScore { get; set; }

    public int ResearchScore { get; set; }

    public int AdministrationScore { get; set; }

    public int ContributionScore { get; set; }

    public string EvidenceSummary { get; set; } = string.Empty;

    public string FacultyRemarks { get; set; } = string.Empty;

    public string ReviewerRemarks { get; set; } = string.Empty;

    public string Status { get; set; } = "Draft";

    public DateTime? SubmittedOn { get; set; }
}
